src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/DataprocConfig.scala

```diff
@@ -1,25 +1,14 @@
 package org.broadinstitute.dsde.workbench.leonardo.config
 
-case class DataprocConfig(applicationName: String,
-                          serviceAccount: String,
-                          dataprocDefaultRegion: String,
-                          leoGoogleBucket: String,
-                          dataprocDockerImage: String,
-                          jupyterProxyDockerImage: String,
-                          clusterFirewallRuleName: String,
-                          clusterFirewallVPCNetwork: String,
-                          clusterNetworkTag: String,
-                          configFolderPath: String,
-                          initActionsScriptName: String,
-                          clusterDockerComposeName: String,
-                          serviceAccountPemName: String,
-                          jupyterServerCrtName: String,
-                          jupyterServerKeyName: String,
-                          jupyterRootCaPemName: String,
-                          jupyterRootCaKeyName: String,
-                          jupyterProxySiteConfName: String,
-                          clusterUrlBase: String,
-                          jupyterServerName: String,
-                          proxyServerName: String,
-                          jupyterInstallExtensionScript: String,
-                          userServiceAccountCredentials: String)
\ No newline at end of file
+import org.broadinstitute.dsde.workbench.leonardo.model.{GoogleProject, GoogleServiceAccount}
+
+case class DataprocConfig(
+                           applicationName: String,
+                           serviceAccount: GoogleServiceAccount,
+                           dataprocDefaultRegion: String,
+                           leoGoogleBucket: GoogleProject,
+                           dataprocDockerImage: String,
+                           jupyterProxyDockerImage: String,
+                           clusterUrlBase: String,
+                           jupyterServerName: String,
+                           proxyServerName: String)
```

Comment:
should the jupyter and proxy lines be in here or in the `proxy` block? i like how you've broken out the blob of cluster resources but the difference between what goes in dataproc vs proxy seems less clear to me

Reply:
Good question. I think `ProxyConfig` is for stuff that the Leo akka-http proxy needs. `DataprocConfig` is stuff that Leo needs to make a dataproc cluster. `jupyterServerName` and `proxyServerName` are actually names of the _docker containers_ of jupyter/apache that live on the cluster master node. So I think they belong in `DataprocConfig`. 

Actually it could be argued that `firewallRuleName`/`firewallVPCNetwork`/`networkTag` belong in `DataprocConfig` because it's information needed at cluster creation time, not by the akka-http proxy. But it's a grey area.